instrumentation/servlet/servlet-5.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v5_0/async/AsyncStartAdvice.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.servlet.v5_0.async;
+
+import static io.opentelemetry.instrumentation.servlet.jakarta.v5_0.JakartaServletHttpServerTracer.tracer;
+
+import io.opentelemetry.javaagent.instrumentation.api.CallDepthThreadLocalMap;
+import jakarta.servlet.AsyncContext;
+import jakarta.servlet.http.HttpServletRequest;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.implementation.bytecode.assign.Assigner;
+
+public class AsyncStartAdvice {
+  @Advice.OnMethodEnter(suppress = Throwable.class)
+  public static void startAsyncEnter() {
+    // This allows to detect the outermost invocation of startAsync in method exit
```

Comment:
why is this important? (since checking `isAsyncListenerAttached` before calling `attachAsyncListener` below already)

Reply:
Working only with the outermost request makes sense because wrappers/subclasses can both wrap the `AsyncContext` from delegates or superclasses and intercept attribute handling. Only manipulating the final request and `AsyncContext` that the application code also sees is in my opinion the safest option. Since exit advices are invoked in reverse order, then omitting that check would make it try to add the listener to the innermost request instead.